agrirouter-sdk-dotnet-standard-impl/Service/Common/MqttMessagingService.cs

```diff
@@ -62,5 +63,39 @@ public MessagingResult Send(MessagingParameters messagingParameters)
             return new MessagingResultBuilder().WithApplicationMessageId(messagingParameters.ApplicationMessageId)
                 .Build();
         }
+
+        /// <summary>
+        ///     Send message to the AR using the given message parameters.
+        /// </summary>
+        /// <param name="messagingParameters">Messaging parameters.</param>
+        /// <returns>-</returns>
+        /// <exception cref="CouldNotSendMqttMessageException">Will be thrown if the message could not be send.</exception>
+        public async Task<MessagingResult> SendAsync(MessagingParameters messagingParameters)
+        {
+            var messageRequest = new MessageRequest
+            {
+                SensorAlternateId = messagingParameters.OnboardResponse.SensorAlternateId,
+                CapabilityAlternateId = messagingParameters.OnboardResponse.CapabilityAlternateId,
+                Messages = new List<Api.Dto.Messaging.Inner.Message>()
+            };
+
+            foreach (var message in messagingParameters.EncodedMessages.Select(encodedMessage =>
+                new Api.Dto.Messaging.Inner.Message
+                    {Content = encodedMessage, Timestamp = UtcDataService.NowAsUnixTimestamp()}))
+                messageRequest.Messages.Add(message);
+
+            var messagePayload = JsonConvert.SerializeObject(messageRequest);
+
+            var mqttMessage = new MqttApplicationMessageBuilder()
+                .WithTopic(messagingParameters.OnboardResponse.ConnectionCriteria.Measures)
+                .WithPayload(messagePayload)
+                .WithExactlyOnceQoS()
+                .WithRetainFlag()
+                .Build();
+
+            await _mqttClient.PublishAsync(mqttMessage, CancellationToken.None);
+
+            return new MessagingResultBuilder().WithApplicationMessageId(messagingParameters.ApplicationMessageId).Build();
+        }
     }
```

Comment:
Same suggestion as for the HTTP Service.

Reply:
Fixes